### DISCUSSION TITLE
Release v0.0.0

### DISCUSSION THREAD

**Discussion by mayank1513** [react18-tools/esbuild-plugin-react18#5] [Show and tell]
## What's Changed
* Setup example package by @mayank1513 in https://github.com/mayank1513/esbuild-plugin-react18/pull/1
* Docs by @mayank1513 in https://github.com/mayank1513/esbuild-plugin-react18/pull/2
* add tests by @mayank1513 in https://github.com/mayank1513/esbuild-plugin-react18/pull/3
* Plugin by @mayank1513 in https://github.com/mayank1513/esbuild-plugin-react18/pull/4

## New Contributors
* @mayank1513 made their first contribution in https://github.com/mayank1513/esbuild-plugin-react18/pull/1

**Full Changelog**: https://github.com/mayank1513/esbuild-plugin-react18/commits/v0.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mayank1513/esbuild-plugin-react18/releases/tag/v0.0.0'>Release v0.0.0</a>.</em>

**Comment by mctrafik** [comment]
What does this plugin actually do? I don't see it use any of the react packages, so it doesn't set anything up. If I don't use NextJS, or Remix... what use is this?!?

**Reply by mayank1513** [reply]
The basic goal of this plugin is to make your library compatible with react 18 RSCs. Additionally, it automatically removes duplicate imports, cleans up testids etc. from the build output, etc. There are hooks to further configure your build if you wish to.

**Reply by mayank1513** [reply]
This plugin is part of build tool. So it will always come as a devDependency. npm etc. track dependencies and not devDeps

**Comment by mctrafik** [comment]
Thank you for that explanation. So IIUC, if there's an esbuild step in RSC
enabled react 18 app, this will make it pass through esbuild so that the
next tool can pick it up. Gotcha.

**Comment by mayank1513** [comment]
This is designed primarily for library builders. I don't understand your
question.

Mayank Kumar Chaudhari
Software Architect (Focus Front End) | Microsoft Certified Expert
https://mayank-chaudhari.vercel.app/
